### PR TITLE
FIX trial rejection criteria computed by Autoreject module.

### LINKED ISSUE #206: Fix autoreject picks
Restrict autoreject trial rejection criteria to MEEG channels only.

### PR DESCRIPTION
**Pull request by ktavabi**
@larsoner please review. Fixes #206

### CODE REVIEW

File: examples/funloc/analysis_fun.py [hunk 1]
```diff
@@ -87,7 +87,8 @@
 params.proj_nums = [[1, 1, 0],  # ECG
                     [1, 1, 2],  # EOG
                     [0, 0, 0]]  # Continuous (from ERM)
-params.autoreject_thresholds = True  # Set to True to use Autoreject module to set global epoch rejection thresholds  # noqa
+params.autoreject_thresholds = True  # Set to True to use Autoreject module to set global epoch rejection thresholds
+params.autoreject_eog = False  # Set to True to use EOG Autoreject criterion to reject trials based on EOG
```
Comment (larsoner) [inline examples/funloc/analysis_fun.py]: I'd rather do `params.autoreject_types` that defaults to `('meg', 'eeg')` (and you can do `('meg', 'eeg', 'eog')` if you want)

File: mnefun/_mnefun.py [hunk 2]
```diff
@@ -1562,6 +1568,14 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
                                  preload=True, decim=decim[si])
             new_dict = get_rejection_threshold(temp_epochs)
             use_reject, use_flat = _restrict_reject_flat(new_dict, p.flat, raw)
+            if p.autoreject_eog:
+                msg = 'You have chosen to reject trials based on criterion computed ' \
+                      'using Autoreject module.\nThis option may result in excessive ' \
+                      'trial rejection. Make sure to check your data!'
+                warnings.warn(msg)
```
Comment (larsoner) [inline mnefun/_mnefun.py]: I don't think we need to warn if the default is `('meg', 'eeg')`

**Comment by ktavabi** [comment]
@larsoner is this ready for merge?

File: mnefun/_mnefun.py [hunk 3]
```diff
@@ -1561,7 +1566,10 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
                                  proj=True, reject=None, flat=None,
                                  preload=True, decim=decim[si])
             new_dict = get_rejection_threshold(temp_epochs)
-            use_reject, use_flat = _restrict_reject_flat(new_dict, p.flat, raw)
+            use_reject = dict()
+            use_reject.update((k, new_dict[k]) for k in p.autoreject_types)
```
Comment (larsoner) [inline mnefun/_mnefun.py]: FYI you can directly do `use_reject = dict((k, new_dict[k]) for k in p.autoreject_types)`

**Pull request merged**